src/main/java/org/opentripplanner/routing/algorithm/strategies/street_heuristic/StreetRWH.java

```diff
@@ -0,0 +1,65 @@
+package org.opentripplanner.routing.algorithm.strategies.street_heuristic;
+
+import org.opentripplanner.routing.algorithm.strategies.RemainingWeightHeuristic;
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.routing.core.State;
+
+import java.util.Arrays;
+import java.util.PriorityQueue;
+
+public class StreetRWH implements RemainingWeightHeuristic {
+
+    private static final float INITIAL_WEIGHT = 180.0f;
+
+    private StreetHeuristicData data;
+
+    private boolean[][] visited;
+
+    private Point destination;
+
+    @Override
+    public void initialize(RoutingRequest options, long abortTime) {
+        this.data = options.rctx.graph.streetHeuristicData;
+        visited = new boolean[data.getBoundaries().getHeight()][data.getBoundaries().getWidth()];
+        destination = data.mapToPoint(options.to); // TODO outside boundaries
+    }
+
+    @Override
+    public double estimateRemainingWeight(State s) {
+        Arrays.stream(visited).forEach(array -> Arrays.fill(array, false));
+
+        PriorityQueue<PointWithWeight> queue = new PriorityQueue<>();
+        Point source = data.mapToPoint(s.getVertex());
+        if (!data.getBoundaries().contains(source)) {
+            return 0.; // TODO euclidean
+        }
+        queue.add(new PointWithWeight(data.mapToPoint(s.getVertex()), INITIAL_WEIGHT));
```

Comment:
add TODO to set INITIAL_WEIGHT based on data

Reply:
done